gradio/components/chatbot.py

```diff
@@ -185,10 +187,17 @@ def _preprocess_chat_messages(
         elif isinstance(chat_message, str):
             return chat_message
         elif isinstance(chat_message, ComponentMessage):
-            component = import_component_and_data(chat_message.component.capitalize())
+            capitalized_component = (
+                chat_message.component.upper()
+                if chat_message.component in ("json", "html")
+                else chat_message.component.capitalize()
+            )
+            component = import_component_and_data(capitalized_component)
             if component is not None:
                 instance = component()  # type: ignore
-                if issubclass(instance.data_model, GradioModel):
+                if not instance.data_model:
+                    payload = chat_message.value
+                elif issubclass(instance.data_model, GradioModel):
```

Comment:
`issubclass` requires the first arg to be a class and if there is no `data_model` then this isn't the case, so handled this earlier.